File: EPs/EP_oo/scripts/bateria.py
class Bateria():
    def __init__(self, mah, tempoDeCarregamento):
        self.mah = mah
        self.tempoDeCarregamento = tempoDeCarregamento
        self.carga = 0
        self.uso
        self.carregavel

    def carregar(self, tempo):
        if self.carregavel == True:
            if self.carga == self.mah:
                print("A bateria ja esta carregada!")
                return False
            else:
                print("A bateria foi carregada com sucesso!")
                self.carga += (self.mah * tempo) / self.tempoDeCarregamento
                return True
        else:
            print("A bateria nao pode ser carregada!")
            return False

    def usar(self, tempo):
        # tempoDeCarregamento == tempoDeVoo
        consumo = (self.mah * tempo) / self.tempoDeCarregamento
        if self.carga > consumo:
            self.carga -= consumo
            return True
        else:
            print("A bateria nao aguenta ser usada por esse intervalo de tempo!")
            return False

    def calcula_tempo_de_voo(self):
        return (self.carga * self.tempoDeCarregamento) / self.mah

    def status(self):
        print("Bateria com capacidade " + str(self.mah) +
              " e tempo de carregamento " + str(self.tempoDeCarregamento))
        print("Com carga atual: " + str(self.carga))
        print("Tempo de voo disponivel: " + str(self.calcula_tempo_de_voo()))
        print("Esta sendo usada: " + "sim" if self.uso == True else "nao")
        print("Pode ser carregada no momento: " + str(self.carregar(0)))

File: EPs/EP_oo/src/drone/Bateria.cpp
#include <iostream>
#include "Bateria.hpp"
using namespace std;

Bateria::Bateria(int mah, int tempoDeCarregamento)
{
  this->mah = mah;
  this->tempoDeCarregamento = tempoDeCarregamento;
}

Bateria::~Bateria() {}

bool Bateria::carregar(int tempo)
{
  if (this->getCarregavel() == true)
  {
    if (carga == this->mah)
    {
      cout << "A bateria ja esta carregada!" << endl;
      return false;
    }
    else
    {
      cout << "A bateria foi carregada com sucesso!" << endl;
      this->carga += (this->mah * tempo) / this->tempoDeCarregamento;
      return true;
    }
  }
  else
  {
    cout << "A bateria nao pode ser carregada!" << endl;
    return false;
  }
}

bool Bateria::usar(int tempo)
{
  int consumo_de_bateria = (this->mah * tempo) / this->tempoDeCarregamento;
  if (this->carga >= consumo_de_bateria && this->carga != 0 && this->uso == true)
  {
    this->carga -= consumo_de_bateria;
    return true;
  }
  else
  {
    cout << "A bateria nao aguenta ser usada por esse intervalo de tempo!" << endl;
    return false;
  }
}

int Bateria::calculaTempoDeVoo()
{
  return (this->carga * this->tempoDeCarregamento) / this->mah;
}

int Bateria::getCarga()
{
  return this->carga;
}

int Bateria::getTempoDeCarregamento()
{
  return this->tempoDeCarregamento;
}

int Bateria::getMah()
{
  return this->mah;
}

bool Bateria::getUso()
{
  return this->uso;
}

bool Bateria::getCarregavel()
{
  return this->carregavel;
}

void Bateria::setUso(bool uso)
{
  this->uso = uso;
}

void Bateria::setCarregavel(bool carregavel)
{
  this->carregavel = carregavel;
}

void Bateria::status()
{
  cout << "Bateria com capacidade " << this->mah << " e tempo de carregamento " << this->tempoDeCarregamento << endl;
  cout << "Com carga atual: " << this->carga << endl;
  cout << "Tempo de voo disponivel: " << this->calculaTempoDeVoo() << endl;
  if (this->uso)
  {
    cout << "Esta sendo usada: sim." << endl;
  }
  else
  {
    cout << "Esta sendo usada: nao." << endl;
  }

  if (this->carregavel)
  {
    cout << "Pode ser carregada no momento: Sim." << endl;
  }
  else
  {
    cout << "Pode ser carregada no momento: Nao." << endl;
  }
}